src/Materials/babylon.materialHelper.ts

```diff
@@ -298,48 +317,39 @@
                     }
                 }
                 effect.setTexture("shadowSampler" + lightIndex, shadowGenerator.getShadowMapForRendering());
-                effect.setFloat3("shadowsInfo" + lightIndex, shadowGenerator.getDarkness(), shadowGenerator.blurScale / shadowGenerator.getShadowMap().getSize().width, shadowGenerator.depthScale);
+                light._uniformBuffer.updateFloat3(useUbo ? "shadowsInfo" : "shadowsInfo" + lightIndex, shadowGenerator.getDarkness(), shadowGenerator.blurScale / shadowGenerator.getShadowMap().getSize().width, shadowGenerator.depthScale);
             }
 
             return depthValuesAlreadySet;
         }
 
         public static BindLightProperties(light: Light, effect: Effect, lightIndex: number): void {
-            if (light.getTypeID() === 0) {
-                // Point Light
-                light.transferToEffect(effect, "vLightData" + lightIndex);
-            } else if (light.getTypeID() === 1) {
-                // Directional Light
-                light.transferToEffect(effect, "vLightData" + lightIndex);
-            } else if (light.getTypeID() === 2) {
-                // Spot Light
-                light.transferToEffect(effect, "vLightData" + lightIndex, "vLightDirection" + lightIndex);
-            } else if (light.getTypeID() === 3) {
-                // Hemispheric Light
-                light.transferToEffect(effect, "vLightData" + lightIndex, "vLightGround" + lightIndex);
-            }
+            light.transferToEffect(effect, lightIndex);
         }
 
         public static BindLights(scene: Scene, mesh: AbstractMesh, effect: Effect, defines: MaterialDefines, maxSimultaneousLights = 4) {
             var lightIndex = 0;
             var depthValuesAlreadySet = false;
 
             for (var light of mesh._lightSources) {
+                var useUbo = light._uniformBuffer.useUbo;

```

Comment:
This should be handled by the uniformBuffer itself
 and then call could be like:
light._uniformBuffer.updateColor4("vLightDiffuse", lightIndex, Tmp.Color3[0], light.range);

where lightIndex is used if there is no UBO